rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -922,6 +922,8 @@ extern "C" rmw_ret_t rmw_init(const rmw_init_options_t * options, rmw_context_t
     if (context->options.security_options.enforce_security == RMW_SECURITY_ENFORCEMENT_ENFORCE) {
       return RMW_RET_ERROR;
     }
+    // No security enforced, lack of support is not an error.
+    rmw_reset_error();
```

Comment:
I think setting the error state is a bug anyway, fixing it like this feels a tad kludgey. I instead suggest surrounding the generation of the error message with a `if (security_options->security_root_path != nullptr)`. If `security_root_path == nullptr` I'd just return OK, since nothing was requested and we did nothing.